Modules/Filtering/LabelMap/test/itkBinaryFillholeImageFilterTest1.cxx

```diff
@@ -76,3 +67,34 @@ itkBinaryFillholeImageFilterTest1(int argc, char * argv[])
   std::cout << "Test finished." << std::endl;
   return EXIT_SUCCESS;
 }
+} // namespace
+
+int
+itkBinaryFillholeImageFilterTest1(int argc, char * argv[])
+{
+  if (argc != 5)
+  {
+    std::cerr << "Missing Parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv)
+              << " inputFileName outputFileName fullyConnected foregroundValue" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using ImageType = itk::Image<short, 2>;
```

Comment:
Not sure why the dimensionality is hard-coded here if the purpose is to test this with 3D data. Maybe it should be provided as an input argument to the test?

Reply:
This image type is only used to instantiate the reader which will determine the on-disk dimension. The `DoIt` has the correct dimension.

Reply:
Maybe it is too long since I have not actually read our docs or actively developed, but not sure if I understand: then why is the reader templated? Why don't we put the dimension to e.g. 1 always, regardless of the dimensionality of the image?

Reply:
> Removing the explicit dimension specification might be better?

If it does not have a function, then to me it does not help to have the dimension specification in any ways.

I will need to revise my knowledge about the toolkit if it does not have a function; I cannot review the PR.

Reply:
This approach is inspired by: https://github.com/Slicer/Slicer/blob/5e45824d5cc654a8277205921fb4a1ed129167bb/Base/CLI/itkPluginUtilities.h#L17-L30

Reply:
The way I sometimes get the dimension of a file is just with the ImageIOBase:
https://github.com/SimpleITK/SimpleITK/blob/46bf97cf954646e284a8ebbc6863ac8b2c5465fe/Code/IO/src/sitkImageReaderBase.cxx#L225